localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp

```diff
@@ -184,112 +182,67 @@ NDTScanMatcher::NDTScanMatcher(const rclcpp::NodeOptions & options)
-  }
-  if (
-    state_ptr_->count("skipping_publish_num") &&
-    std::stoi((*state_ptr_)["skipping_publish_num"]) > 1 &&
-    std::stoi((*state_ptr_)["skipping_publish_num"]) < 5) {
-    diag_status_msg.level = diagnostic_msgs::msg::DiagnosticStatus::WARN;
-    diag_status_msg.message += "skipping_publish_num > 1. ";
-  }
-  if (
-    state_ptr_->count("skipping_publish_num") &&
-    std::stoi((*state_ptr_)["skipping_publish_num"]) >= 5) {
-    diag_status_msg.level = diagnostic_msgs::msg::DiagnosticStatus::ERROR;
-    diag_status_msg.message += "skipping_publish_num exceed limit. ";
-  }
-  if (
-    state_ptr_->count("nearest_voxel_transformation_likelihood") &&
-    std::stod((*state_ptr_)["nearest_voxel_transformation_likelihood"]) <
-      param_.score_estimation.converged_param_nearest_voxel_transformation_likelihood) {
-    diag_status_msg.level = diagnostic_msgs::msg::DiagnosticStatus::WARN;
-    diag_status_msg.message += "NDT score is unreliably low. ";
-  }
-  if (
-    state_ptr_->count("execution_time") && std::stod((*state_ptr_)["execution_time"]) >=
-                                             param_.validation.critical_upper_bound_exe_time_ms) {
-    diag_status_msg.level = diagnostic_msgs::msg::DiagnosticStatus::WARN;
-    diag_status_msg.message +=
-      "NDT exe time is too long. (took " + (*state_ptr_)["execution_time"] + " [ms])";
-  }
-  // Ignore local optimal solution
-  if (
-    state_ptr_->count("is_local_optimal_solution_oscillation") &&
-    std::stoi((*state_ptr_)["is_local_optimal_solution_oscillation"])) {
-    diag_status_msg.level = diagnostic_msgs::msg::DiagnosticStatus::OK;
-    diag_status_msg.message = "local optimal solution oscillation occurred";
-  }
-
-  diagnostic_msgs::msg::DiagnosticArray diag_msg;
-  diag_msg.header.stamp = this->now();
-  diag_msg.status.push_back(diag_status_msg);
-
-  diagnostics_pub_->publish(diag_msg);
+  map_update_module_->callback_timer(is_activated_, latest_ekf_position_);
 }
 
-void NDTScanMatcher::callback_timer()
+void NDTScanMatcher::callback_initial_pose(
+  const geometry_msgs::msg::PoseWithCovarianceStamped::ConstSharedPtr initial_pose_msg_ptr)
 {
-  if (!is_activated_) {
-    return;
-  }
-  std::lock_guard<std::mutex> lock(latest_ekf_position_mtx_);
-  if (latest_ekf_position_ == std::nullopt) {
-    RCLCPP_ERROR_STREAM_THROTTLE(
-      this->get_logger(), *this->get_clock(), 1000,
-      "Cannot find the reference position for map update. Please check if the EKF odometry is "
-      "provided to NDT.");
-    return;
-  }
-  // continue only if we should update the map
-  if (map_update_module_->should_update_map(latest_ekf_position_.value())) {
-    RCLCPP_INFO(this->get_logger(), "Start updating NDT map (timer_callback)");
-    map_update_module_->update_map(latest_ekf_position_.value());
-  }
+  diagnostics_initial_pose_->clear();
+
+  set_initial_pose(initial_pose_msg_ptr);
+
+  diagnostics_initial_pose_->publish();
```

Comment:
I like this style.
I think it's a good idea to construct all callbacks with three lines like this to create a unified style.

```
diagnostics_xxx_->clear();
main_process();
diagnostics_xxx_->publish();
```

callbacks

* callback_timer() : OK
* callback_initial_pose() : OK
* callback_regularization_pose() : OK
* callback_sensor_points () 
 : The style may be a little different　https://github.com/autowarefoundation/autoware.universe/blob/5f8391846b7922689f7b371b5bc731c746cf16fc/localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp#L265-L309
* MapUpdateModule::callback_timer() : The style may be a little different https://github.com/autowarefoundation/autoware.universe/blob/5f8391846b7922689f7b371b5bc731c746cf16fc/localization/ndt_scan_matcher/src/map_update_module.cpp#L67-L74